### PR TITLE
SPLICE-888 Removing custom Predicate Wrapper, predicate interaction i…

### PR DESCRIPTION
**Pull request by jleach4**
Removal of custom predicate wrapper, predicate application inside encoding, and Lazy data value descriptor implementation.

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/GenericQualifier.java [hunk 1]
```diff
@@ -235,4 +233,9 @@ public String toString()
 			return "";
 		}
 	}
+
+	@Override
+	public int getVariantType() {
```
Comment (jeffcunningham) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/GenericQualifier.java]: should be enum.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/LazyDataValueFactory.java [hunk 2]
```diff
@@ -15,263 +15,66 @@
...
-        if (dataValue == null)
-        {
-            try
-            {
-                return new LazyTimestamp(new SQLTimestamp((Timestamp) null));
+        public NumberDataValue getDecimalDataValue(Long value, NumberDataValue previous) throws StandardException {
+                if(previous != null && previous instanceof SQLDecimal){
+                        previous.setValue(value);
+                    }else{
+                        previous = new SQLDecimal(BigDecimal.valueOf(value.longValue()));
```
Comment (jeffcunningham) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/LazyDataValueFactory.java]: still needed?

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/LazyDataValueFactory.java [hunk 3]
```diff
@@ -15,263 +15,66 @@
...
+                    }
+        
+                        return previous;
             }
-            catch( StandardException se)
-            {
-                return null;
+    
+                public NumberDataValue getDecimalDataValue(String value) throws StandardException {
+                return new SQLDecimal(value);
```
Comment (jeffcunningham) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/LazyDataValueFactory.java]: Still needed?

File: splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/SpliceTransactionFactory.java [hunk 4]
```diff
@@ -253,4 +258,33 @@ public void stop(){
     public FileResource getFileHandler(){
         return fileHandler;
     }
+
+    /**
+     * Privileged startup. Must be private so that user code
+     * can't call this entry point.
+     */
+    private  static  Object bootServiceModule
```
Comment (jeffcunningham) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/store/access/SpliceTransactionFactory.java]: Remove

File: splice_machine/src/main/java/com/splicemachine/derby/stream/iapi/ScanSetBuilder.java [hunk 5]
```diff
@@ -48,8 +46,12 @@
 
     ScanSetBuilder<V> transaction(TxnView txn);
 
+    ScanSetBuilder<V> optionalProbeValue(DataValueDescriptor optionalProbeValue);
```
Comment (scottfines) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/iapi/ScanSetBuilder.java]: Could we get some documentation about what this does?

File: splice_machine/src/main/java/com/splicemachine/derby/utils/Scans.java [hunk 6]
```diff
@@ -313,4 +271,136 @@ private static boolean isEmpty(int[] array) {
...
+            }
+            row_qualifies =
+                    columnValue.compare(
+                            q.getOperator(),
+                            probeValue==null || i!=0?q.getOrderable():probeValue,
+                            q.getOrderedNulls(),
+                            q.getUnknownRV());
+            if (q.negateCompareResult())
+                row_qualifies = !row_qualifies;
+//            System.out.println(String.format("And Clause -> value={%s}, operator={%s}, orderable={%s}, " +
```
Comment (scottfines) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/Scans.java]: Let's remove the commented out debugging stuff, please :)

**Comment by jeffcunningham** [comment]
+1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java [hunk 7]
```diff
@@ -92,10 +92,11 @@ public void boot(boolean create,Properties startParams) throws StandardException
             System.setProperty("com.splicemachine.enableLegacyAsserts",Boolean.TRUE.toString());
             System.setProperty("derby.language.logStatementText",Boolean.toString(true));
         }
-        if(config.debugDumpClassFile()) {
+//        if(config.debugDumpClassFile()) {
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java]: This should be uncommented, right?

**Comment by dgomezferro** [comment]
+1

**Pull request closed without merging**